yamlpatch/types.go

```diff
@@ -28,6 +28,9 @@ type Operation struct {
 	Path  Path        `json:"path" yaml:"path"`
 	From  Path        `json:"from,omitempty" yaml:"from,omitempty"`
 	Value interface{} `json:"value,omitempty" yaml:"value,omitempty"`
+
+	// If not empty, will be inserted as a head comment above this node
```

Comment:
Can you include which operations support adding the comment? Looks like add/replace/test. Should we error for the others if the incorrectly specify a comment?

Maybe also worth mentioning that this is not part of RFC6902